WeatherInfoWidget/force-app/main/default/classes/WeatherCalloutExtension.cls

```diff
@@ -0,0 +1,46 @@
+/*************************************************************************************
+ * @Name         : WeatherCalloutExtension.apxc
+ * @Description  : Standard Controller Extension to send an HTTP callout and set the results to a wrapper class
+ * @Created By   : Berke Bakkaloğlu
+ * @Created Date : Sep 01, 2021
+ *************************************************************************************/
+public class WeatherCalloutExtension {
+    public WeatherWrapper weatherData {get; set;}
+    private final String API_KEY = '0dd7ab53306406da32958d6fd85f889f';
+    private final Account account;
+    
+    public WeatherCalloutExtension(ApexPages.StandardController stdController) {
+        if (!Test.isRunningTest()) {
+            stdController.addFields(new List<String> {'BillingCity'});
+        }
+        this.account = (Account) stdController.getRecord();
+        this.weatherData = doWeatherCallout();
+    }
+    
+    /*************************************************************************************
+    * @Name         : doWeatherCallout
+    * @Description  : Sends an HTTP callout and deserializes the JSON response into a wrapper class
+    * @Created By   : Berke Bakkaloğlu
+    * @Created Date : Sep 01, 2021
+    * @Return	WeatherWrapper	Deserialized version of the callout response
+    *************************************************************************************/
+    private WeatherWrapper doWeatherCallout() {
+        HttpRequest request = new HttpRequest();
+        HttpResponse response = new HttpResponse();
+        Http http = new Http();
+        
+        request.setEndpoint('https://api.openweathermap.org/data/2.5/weather?q=' + account.BillingCity + '&appid=' + API_KEY);
+        request.setMethod('GET');
+        
+        response = http.send(request);
+        
+        if (response.getStatusCode() == 200) {
```

Comment:
I suggest to add a bit more conditions here
if status >= 200 AND < 300 --- OK
if status >=400 AND < 500 --- bad request
if status >=500 and < 600 --- some server error

try to output different error messages on a screen. You can use <apex:pageMessages /> in VF page